docs/transaction-api.asciidoc

```diff
@@ -64,14 +64,58 @@ Arguments:
 
 * `options` - The following options are supported:
 
+** `blocked` - Blocks the associated transaction from ending till this span is ended. Blocked spans
+    automatically creates a task. Defaults to false
+
 ** `parentId` - Parent id associated with the new span. Defaults to current transaction id
 
-** `sync` - Denotes if the span is blocking (sync) or non-blocking(async). 
+** `sync` - Denotes if the span is synchronous or asynchronous.
 
 When a span is started it will measure the time until <<span-end,`span.end()`>> is called.
 
 See <<span-api,Span API>> docs for details on how to use custom spans.
 
+[float]
+[[transaction-add-task]]
+==== `transaction.addTask([taskId])`
+
+[source,js]
+----
+const taskId = transaction.addTask('async-task')
+----
+
+Starts a new task for the current transaction and returns the new task id. The transaction does not end till all the tasks are
+removed from the transacion.
+
+Arguments:
+
+* `name` - Optional. The name of the task (string). Defaults to a random id.
+
+Tasks are pretty new and its available only on the RUM agent. Most of the time, the users shouldn't need to create tasks as
+custom transactions created by the user ends only when `transaction.end` method is called. However in the case of managed transactions like page-load, route-change and other user created ones by setting `managed: true` flag, the transaction might get closed automatically prior to the user's expectation once all the associated tasks are completed. Please check the <<custom-managed-transactions, custom transactions>> guide to understand how you can control the behaviour of transaction using tasks.
+
+
+NOTE: This API will be removed in the next major version, Please use the `startSpan` method with `blocking: true`.
```

Comment:
Let's remove this for now, we can add if we decide to remove the task API.

Reply:
Check my comment here, https://github.com/elastic/apm-agent-rum-js/pull/866#discussion_r469817535

I agree with you that users might not need to measure operations always, But WDYT about an alternative idea where the users will do

```
transaction.block(true)

transaction.block(false)
```

With this
+ we remove a new concept called Task
+ Easier for us to manage as well, We will internally still keep tasks and we can also use this signal for managed transactions.